### DISCUSSION TITLE
Docker: FileUpload - Buffer Full

### DISCUSSION THREAD

**Discussion by ganonym** [pocketbase/pocketbase#5232] [Q&A] [answered]
Hello everyone,

I currently have a problem uploading files both via the admin interface and the PB JS-SDK. As soon as the file has a size of more than ~40MB, the following error appears in the interface with a corresponding log: 

![image](https://github.com/user-attachments/assets/154d0733-362b-4a05-a4c5-ef6dca55d224)

![image](https://github.com/user-attachments/assets/0d3a8d84-49c9-4ec2-b673-d504c0ca74ba)

The pocket base instance runs in a simple Docker container. After a few tests, I noticed that the error does not occur with a local server instance on Windows or Linux and that files are uploaded correctly. However, as soon as a Docker instance is used, the error described above occurs. 

Is this initially due to Docker? Do I have to specify something special in the Dockerfile so that the error no longer occurs? Or is the pocketbaseserver in the container not handling the request correctly?

**Comment by ganigeorgiev** [comment, marked as answer]
I don't think your issue is PocketBase related (especially if it works fine outside of the container).

I don't specifically recognize the above error, but `multipart/form-data` errors in a container env could happen mostly because of:

- reverse proxy (nginx, apache, caddy, etc.) body size and read limits
- invalid configured or limited `/tmp` mount size (this is because uploaded files larger than 32MB are temporary stored on the disk when parsing the form data)
  _If you are not sure, you can run from your container something like `docker run -ti yourcontainer df -h /tmp` to check the available free space dedicated for the `/tmp` dir; if it is something small you can try increasing it (no matter whether you are using tmpfs/memory or a real disk space)._

Another thing that can help you with the debugging is to start the executable with `--dev` and see if there any more detailed logs in the stderr output of your container.

**Reply by ganonym** [reply]
Thank you very much for your help!

The second point has fixed my error.
In the Dockerfile I had used the _scratch_ image, which could not have a `/tmpfs` directory. After I switched to the _alpine_ image, the upload of files went through completely again.
